File: java/Refactor1/src/main/java/no/bouvet/solid/srpdip/Deduplicator.java
package no.bouvet.solid.srpdip;

import java.util.HashMap;
import java.util.Map;

import no.bouvet.solid.srpdip.messageinterface.ResponseMessage;

public class Deduplicator {

	private Map<Long, ResponseMessage> processedMessages = new HashMap<>();

	public ResponseMessage getExistingResponseMessage(long requestId) {
		return processedMessages.get(requestId);
	}

	public void storeResponseToMessage(long requestId, ResponseMessage response) {
		processedMessages.put(requestId, response);
	}
}

File: cs/Refactor2/SRP_DI_Workshop/Deduplicator.cs
using System.Collections.Generic;
using SRP_DI_Workshop.ServiceInterface;

namespace SRP_DI_Workshop
{
    public class Deduplicator : IDeduplicator
    {
        private readonly Dictionary<long, ResponseMessage> _processedMessages = new Dictionary<long, ResponseMessage>();

        public ResponseMessage GetExistingResponseMessage(long requestId)
        {
            ResponseMessage responseToPreviousMessage;

            return _processedMessages.TryGetValue(requestId, out responseToPreviousMessage) 
                ? responseToPreviousMessage 
                : null;
        }

        public void StoreResponseToMessage(long requestId, ResponseMessage response)
        {
            _processedMessages[requestId] = response;
        }
    }
}